src/Java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntityTreeFarm.java

```diff
@@ -300,27 +335,27 @@ public void onPostTick(IGregTechTileEntity aBaseMetaTileEntity, long aTick) {
 						long tVoltage = getMaxInputVoltage();
 						byte tTier = (byte) Math.max(1, GT_Utility.getTier(tVoltage));						
 						if (EU.getCharge(invItem) >= tVoltage) {
-							Logger.WARNING("Can drain.");
+							//Logger.WARNING("Can drain.");
 							if (EU.discharge(invItem, (int) tVoltage, -1)) {
-								Logger.WARNING("Drained Power.");
+								//Logger.WARNING("Drained Power.");
 								didElectricDamage = true;
 							}
 							else {
-								Logger.WARNING("Failed when draining Power.");
+								//Logger.WARNING("Failed when draining Power.");
 								this.getBaseMetaTileEntity().disableWorking();
 							}
 						}							
 					}
 				}
-				Logger.WARNING("Drained Power? "+didElectricDamage);
+				//Logger.WARNING("Drained Power? "+didElectricDamage);
 
 
 
 				if (!didElectricDamage && invItem.getItem() instanceof GT_MetaGenerated_Tool) {
 					long aDmg = GT_MetaGenerated_Tool.getToolDamage(invItem);
 					long aDmgMax = GT_MetaGenerated_Tool.getToolMaxDamage(invItem);
 					if (aDmg < aDmgMax && GT_MetaGenerated_Tool.getPrimaryMaterial(invItem) != Materials._NULL) {
-						Logger.WARNING("dmg: "+aDmg+" | max: "+aDmgMax);
+						//Logger.WARNING("dmg: "+aDmg+" | max: "+aDmgMax);
 						GT_MetaGenerated_Tool.setToolDamage(invItem, aDmg+getDamageToComponent(invItem));							
```

Comment:
Probably better to use GT_ModHandler.damageOrDechargeItem()?

Reply:
Not sure what you mean, i'm not used of programming java and/or at gt++ code.
I only deleted a logger here. It looked like something used to debug the machine while code was being written.

Reply:
I’m suggesting you change the way items are damaged to the method I’ve provided. It’s better.

Reply:
I tested that method, minimum damage that can be done is 100.
Tool can't be easily replaced from buses or by pushing into the controller, so I preferred disabling the damage of it completely for now.
After testing on zeta we'll see if it's worth keeping like this or changing it back to damaging the tool.